index.js

```diff
@@ -240,13 +240,19 @@ FSWatcher.prototype._awaitWriteFinish = function(path, threshold, callback) {
   (function awaitWriteFinish (prevStat) {
     fs.stat(fullPath, function(err, curStat) {
       if (err) {
-        delete this._pendingWrites[path];
+        // prevent `unlink` race condition
+        this._getWatchedDir(sysPath.dirname(path))
+          .remove(sysPath.basename(path));
+        if (this._pendingWrites[path]) {  
+          this._pendingWrites[path].cancelWait();
```

Comment:
So we have a race condition between the file polling and the unlink event. one of them should cancel every thing about the file - the (maybe) ongoing timeout, and release the pending [async function](https://github.com/paulmillr/chokidar/blob/master/index.js#L156). In my original code I've relayed on that the unlink event will cancel the wait, and if the polling will come first, it will notice that the file is missing (ENOENT) and do nothing. 

Have you seen a case where the unlink event does not arrive at all?